### PR TITLE
Make repair work

### PR DESCRIPTION
**Pull request by agrover**
See https://github.com/stratis-storage/stratisd/pull/359, these are some dm-rs changes needed for that to work.

### CODE REVIEW

**Comment by agrover** [comment]
@trgill @mulkieran review?

**Comment by mulkieran** [comment]
Other than that we're digging ourselves deeper wrt #52, I have no further remarks.

File: src/segment.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ use types::Sectors;
 use device::Device;
 
 /// struct to represent a continuous set of sectors on a disk
-#[derive(Debug)]
```
Comment (agrover) [inline src/segment.rs]: hmm, I thought https://github.com/stratis-storage/stratisd/pull/414 needed it but it does not! Ok, removing.

**Comment by agrover** [comment]
made recommended changes.

**Pull request merged**